### DISCUSSION TITLE
Use Mitsbua Renderer within Blender?

### DISCUSSION THREAD

**Discussion by matthewdhull** [mitsuba-renderer/mitsuba-blender#49] [Q&A] [answered]
Thank you for providing this useful plug in for Blender! This has significantly increased the ease in which I can construct scenes for Mitsuba.

After installing the latest version of Mitsuba-Blender, I noticed that Mitsuba is 'selectable' as a renderer within Blender.  After importing the scene from the repo `test1.xml`, I attempted to see if I could somehow use the renderer within Mitsuba but couldn't quite figure it out.  

I am familiar with how to render using Mitsuba in Python but hoped to take advantage of the newer functionality if this is the intent.

Can you provide any assistance with this? 

<img width="290" alt="image" src="https://user-images.githubusercontent.com/683979/195459278-2b75c8f6-30f0-430b-b0af-6fa927dc1acb.png">

**Comment by ros-dorian** [comment, marked as answer]
Hello, thank you for the comment!

The rendering inside of Blender is not finished yet but we are working on it. Right now, if you import a Mitsuba scene in Blender, the materials are converted to Cycles shader nodes, so you can use Cycles to preview your scene.

Cheers!

**Reply by matthewdhull** [reply]
Thanks, @ros-dorian ! I see this is now a discussion so I won't open a new topic and we'll just use this thread.

**Comment by xdobetter** [comment]
It's a great work.But I happened error when I use mitsuba(v3.0.1) render scene in Blender(v2.93.6),

> Python: Traceback (most recent call last):
  File "C:\Users\DoBetter\AppData\Roaming\Blender Foundation\Blender\2.93\scripts\addons\mitsuba-blender\engine\final.py", line 42, in render
    filepath = os.path.join(dummy_dir, "scene.xml")
TypeError: scene_to_dict() missing 1 required positional argument: 'window_manager'
location: <unknown location>:-1

I don't know whether no support still mitsuba renderer in blender,but I see a interface in blender,so I do try it.like Cycles,Eevee,Workbench it's Ok.
I can succeed import mitsuba.xml or export mitsuba.xml,but I can't use mitsuba renderer in Blender 2.93,below is more information


![1](https://user-images.githubusercontent.com/26523963/234490550-d9e11cae-d718-4e5b-831c-9d02a45c7dd4.png)


![2](https://user-images.githubusercontent.com/26523963/234490567-c89bc48b-f1ac-4856-a461-a7d4e5512ebe.png)


![3](https://user-images.githubusercontent.com/26523963/234490591-838a1cb4-6906-44a6-a67b-6a0cfbf0f972.png)


![4](https://user-images.githubusercontent.com/26523963/234490608-3202048d-13d2-4f1f-9e83-32d694a5749d.png)

**Comment by NoctisC7** [comment]
Can it have more options and materials in nodes so I could export fully set up scene into mitsuba just like in mitsuba 0.5 addon. Would be nice if it opens a window with some basic controls like to stop render or save image + preview of tiles and rendering. I so much want to try it but its way to much of pain to program every step including materials etc. + it would be nice to have all integrators which made for mitsuba not just path tracing, like I remember having BTPT, PSSMLT, SPPM, CPPM etc cant see them in mitsuba 3. IF anyone could compile even old version like Mitsuba 0.6 for new blender with GUI app like 0.5 I would be so happy. Love mitsuba, thanks for wonderful render engine.

**Comment by AnsonSavage** [comment]
It looks like [this](https://github.com/mitsuba-renderer/mitsuba-blender/pull/121) pull request seeks to be able to render in the Blender viewport :)
